src/TLabel.cpp

```diff
@@ -114,3 +146,60 @@ void TLabel::enterEvent(QEvent* event)
     }
     QWidget::enterEvent(event);
 }
+
+bool TLabel::forwardEventToMapper(QEvent* event)
+{
+    if (event->type() == QEvent::MouseButtonPress || event->type() == QEvent::MouseButtonRelease || event->type() == QEvent::MouseMove)
+    {
+        QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
+        QWidget* qw = qApp->widgetAt(mouseEvent->globalPos());
+
+        if (qw && this->parentWidget()->findChild<QWidget*>(QStringLiteral("mapper"))->isAncestorOf(qw))
```

Comment:
AFAICT `this->` is not needed here... and `parentWidget()->findChild<QWidget*>(QStringLiteral("mapper"))` can return a `nullptr` so that needs to be checked before then testing to see if `isAncestorOf(qw)` can be used with it - or that is how I read it -  so I'd revise this to:

    if (qw && this->parentWidget()->findChild<QWidget*>(QStringLiteral("mapper")) && this->parentWidget()->findChild<QWidget*>(QStringLiteral("mapper"))->isAncestorOf(qw))

But then I think "_rain is wet"_ so what do I know? 😀